modules/core/src/main/java/org/apache/ignite/cache/query/SqlFieldsQuery.java

```diff
@@ -121,6 +125,27 @@ public SqlFieldsQuery setArgs(Object... args) {
     }
 
     /**
+     * Gets the query execution timeout in milliseconds.
+     */
+    public int getTimeout() {
+        return timeout;
+    }
+
+    /**
+     * Sets the query execution timeout. Query will be automatically cancelled if the execution timeout is exceeded.
+     * @param timeout Timeout value.
+     * @param timeUnit Time unit.
+     */
+    public void setTimeout(int timeout, TimeUnit timeUnit) {
+        A.ensure(timeUnit != TimeUnit.MICROSECONDS && timeUnit != TimeUnit.NANOSECONDS,
+            "timeUnit minimal resolution is millisecond.");
+
+        A.ensure(timeout > 0, "timeout value should be positive.");
+
+        this.timeout = (int)TimeUnit.MILLISECONDS.convert(timeout, timeUnit);
```

Comment:
Here we need a conversion to `long` and then overflow check (<= Integer.MAX_VALUE).